week3/vs_game/src/App.js

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import GlobalStyle from './components/GlobalStyle'
+import Game from './components/Game'
+import Home from './components/Main'
+import { Route } from "react-router-dom";
+
+function App() {
+  return (
+    <>
+      <GlobalStyle></GlobalStyle>
+        <Route exact path="/" component={Home}/>
+        <Route path="/game" component={Game}/>
+    </>
+  );
+}
```

Comment:
좀 더 나은 구조로 리팩토링 할 수 있을 것 같아서 순서대로 방법 남겨둘게 ㅎㅎ

**1. index.js에 쓰인 BrowserRouter를 지우고 아래처럼 작성한다.**
```
const root = ReactDOM.createRoot(document.getElementById("root"));
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
```

**2. week3/src/components/common/Router.js를 만들고 아래와 같은 코드를 작성한다.**
```javascript
function Router() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/game" element={<Game />} />
        <Route path="/*" element={<p>Page Not Found</p>} />
      </Routes>
    </BrowserRouter>
  );
}

export default Router;
```
**3. 위에서 만든 Router를 App.js에서 import해서 쓴다.**
```suggestion
function App() {
  return (
    <>
        <GlobalStyle />
        <Router />
    </>
  );
}
```
➡️ 이렇게 하면 BrowserRouter는 index.js에 있고, Route는 App.js에 있어서 라우터 관련된 애들이 흩어져 있는 걸 막을 수 있겠지? 라우터를 한 파일에서 관리하기 더 쉽다는 장점이 있으니까 앞으로 이렇게 구조 잡는 거 추천!

Reply:
와우 안그래도 방금 지연이 코드보고 리뷰남겼는데 !! 내가 라우터 버전 5를 깔아서 Routes가 안먹던데 다시깔면 적용이 되려나??

다음 과제에 라우팅할때 이렇게 해봐야겠다 조언해줘서 고마웝 ~~!!